src/file_operations.rs

```diff
@@ -0,0 +1,97 @@
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+
+enum Op {
+    MkDir(PathBuf),
+    GitInit(PathBuf),
+}
+
+struct FileOperations {
+    root: PathBuf,
+    operations: Vec<Op>,
+}
+
+impl FileOperations {
+    fn rooted_at<P: AsRef<Path>>(path: P) -> FileOperations {
```

Comment:
So `rooted_at` is the constructor for a new instance of this type? I really like the name, but why not `new`?